tests.sh

```diff
@@ -371,43 +413,122 @@ run_mdtest(){
-        exit 1
+        teardown_test
     else
         echo -e "\nSTATUS: MDTEST SUCCESS\n"
     fi
     sleep 5
 }
 
-#Prepare Enviornment
-prepare
+# Get a random server name from the SERVER_HOSTLIST_FILE
+# the "lucky" server name will never be the access point
+function get_doom_server(){
+    local MAX_RETRY_ATTEMPTS=1000
+    local ACESS_POINT=$(cat ${SERVER_HOSTLIST_FILE} | head -1)
 
-test=$1
+    n=1
+    until [ ${n} -ge ${MAX_RETRY_ATTEMPTS} ]
+    do
+        local DOOMED_SERVER=$(shuf -n 1 ${SERVER_HOSTLIST_FILE})
+
+        if [ "${ACESS_POINT}" != "${DOOMED_SERVER}" ]; then
+            break
+        fi
+
+        n=$[${n} + 1]
+    done
+
+    if [ ${n} -ge ${MAX_RETRY_ATTEMPTS} ]; then
+        pmsg "hostlist too small and we have really bad lucky"
+        teardown_test
+    fi
+
+    echo ${DOOMED_SERVER}
+}
 
-echo "###################"
-echo "RUN: $TESTCASE"
-echo "Start Time: $(date)"
-echo "###################"
-
-case $test in
-    IOR)
-        start_server
-        start_agent
-        create_pool
-        create_container
-        run_ior
-        ;;
-    SELF_TEST)
-        start_server
-        dump_attach_info
-        run_self_test
-        ;;
-    MDTEST)
-        start_server
-        start_agent
-        create_pool
-        run_mdtest
-        ;;  
-    *)
-        echo "Unknown test: Please use IOR, SELF_TEST or MDTEST"
-esac
+# Kill one DAOS server randomly selected from the SERVER_HOSTLIST_FILE
+function kill_random_server(){
+    local DOOMED_SERVER=$(get_doom_server)
+    local MAX_RETRY_ATTEMPTS=30
+
+    run_cmd "dmg -o ${DAOS_CONTROL_YAML} pool list"
+    run_cmd "dmg -o ${DAOS_CONTROL_YAML} pool query --pool ${POOL_UUID}"
+    run_cmd "dmg -o ${DAOS_CONTROL_YAML} system query"
+
+    pmsg "Waiting to kill ${DOOMED_SERVER} server in ${WAIT_TIME} seconds..."
+    sleep ${WAIT_TIME}
+    pmsg "Killing ${DOOMED_SERVER} server"
+    run_cmd "ssh ${DOOMED_SERVER} \"pgrep -a ${PROCESSES}\""
+    run_cmd "ssh ${DOOMED_SERVER} \"pkill ${PROCESSES}\""
+    run_cmd "ssh ${DOOMED_SERVER} \"pgrep -a ${PROCESSES}\""
+    pmsg "Killed ${DOOMED_SERVER} server"
+
+    n=1
+    until [ ${n} -ge ${MAX_RETRY_ATTEMPTS} ]
+    do
+        run_cmd "dmg -o ${DAOS_CONTROL_YAML} pool list"
+        run_cmd "dmg -o ${DAOS_CONTROL_YAML} system query"
+        run_cmd "dmg -o ${DAOS_CONTROL_YAML} pool query \
+                 --pool ${POOL_UUID}"
+
+        if echo "${OUTPUT_CMD}" | grep -qE "^Rebuild\sdone"; then
+            break
+        fi
+
+        pmsg "Attempt ${n} failed, retrying in 1 second..."
+        n=$[${n} + 1]
+        sleep 1
+    done
+
+    if [ ${n} -ge ${MAX_RETRY_ATTEMPTS} ]; then
+        pmsg "Failed to rebuild pool ${POOL_UUID}"
+        teardown_test
+      fi
+}
+
+function run_testcase(){
+    #Prepare Enviornment
+    prepare
+
+    echo "###################"
+    echo "RUN: $TESTCASE"
+    echo "Start Time: $(date)"
+    echo "###################"
+
+    case ${test} in
+        STABILIZATION)
```

Comment:
can you rename STABILIZATION to SWIM?
and add a comment "Swim stabilization test by checking server fault detection"